esp-hal/src/gpio/interconnect.rs

```diff
@@ -406,21 +442,66 @@ impl OutputSignal {
             pub fn set_output_high(&mut self, on: bool, _internal: private::Internal);
             pub fn set_drive_strength(&mut self, strength: gpio::DriveStrength, _internal: private::Internal);
             pub fn enable_open_drain(&mut self, on: bool, _internal: private::Internal);
-            pub fn enable_output_in_sleep_mode(&mut self, on: bool, _internal: private::Internal);
             pub fn internal_pull_up_in_sleep_mode(&mut self, on: bool, _internal: private::Internal);
             pub fn internal_pull_down_in_sleep_mode(&mut self, on: bool, _internal: private::Internal);
             pub fn is_set_high(&self, _internal: private::Internal) -> bool;
         }
     }
 }
 
+struct DirectOutputSignal {
```

Comment:
I know this is an internal API, but a small comment here would be helpful. Same for `DirectInputSignal`.

It's a bit unclear to me what this is, judging by the vibe of the whole PR, along with the comments it forces the signals to go through the GPIO matrix, but I would argue that `Direct` naming somewhat contradicts that. Could we clarify this a bit?

Reply:
Yeah I introduced new code to keep supporting the old semantics where possible, then modified the existing code wasn't the best strategy for easy reviews, I guess.

Reply:
Thanks!